File: src/main/java/de/byjoker/myjfql/database/Table.java
package de.byjoker.myjfql.database;

import de.byjoker.myjfql.lang.TableEntryComparator;
import de.byjoker.myjfql.util.Order;

import java.util.Collection;

public interface Table {

    void addEntry(TableEntry tableEntry);

    void removeEntry(String identifier);

    TableEntry getEntry(String identifier);

    Collection<TableEntry> getEntries();

    Collection<TableEntry> getEntries(TableEntryComparator comparator, Order order);

    Collection<String> getStructure();

    void setStructure(Collection<String> structure);

    String getPrimary();

    void setPrimary(String primary);

    Table reformat(TableType type);

    TableType getType();

    String getName();

    void clear();
}

File: src/main/java/de/byjoker/myjfql/database/Table.kt
package de.byjoker.myjfql.database

import com.fasterxml.jackson.annotation.JsonIgnore
import de.byjoker.myjfql.lang.Requirement

interface Table {

    val id: String
    var name: String

    val structure: List<String>
    val primary: String
    val partitioner: String
    val type: TableType
    fun pushEntry(entry: Entry)
    fun getEntry(entryId: String): Entry?
    fun removeEntry(entryId: String)
    fun findEntries(conditions: List<List<Requirement>>, limit: Int): List<Entry>
    fun getEntries(): List<Entry>
    fun clear()

    @get:JsonIgnore
    val databaseId: String

}
